src/Modules/OrchardCore.Commerce.Payment.Stripe/Handlers/SubscriptionStripeWebhookEventHandler.cs

```diff
@@ -54,7 +54,11 @@ public async Task ReceivedStripeEventAsync(Event stripeEvent)
                 var subscriptionPart = new SubscriptionPart();
                 subscriptionPart.UserId.Text = user.UserId;
                 subscriptionPart.Status.Text = SubscriptionStatuses.Active;
-                subscriptionPart.EndDateUtc.Value = invoice.PeriodEnd;
+
+                // invoice.PeriodEnd doesn't show the current period, see Stripe docs:
+                // End of the usage period during which invoice items were added to this invoice. This looks back one
+                // period for a subscription invoice. Use the line item period to get the service period for each price.
```

Comment:
Include a URL to said docs, not just a quote.

```suggestion
                // invoice.PeriodEnd doesn't show the current period, see Stripe docs:
                // https://docs.stripe.com/api/invoices/object#invoice_object-period_end
                // End of the usage period during which invoice items were added to this invoice. This looks back one
                // period for a subscription invoice. Use the line item period to get the service period for each price.
```

Also I'm not sure about this, but it might be better to put the text in quotes?

```suggestion
                // invoice.PeriodEnd doesn't show the current period, see Stripe docs:
                // https://docs.stripe.com/api/invoices/object#invoice_object-period_end
                // "End of the usage period during which invoice items were added to this invoice. This looks back one
                // period for a subscription invoice. Use the line item period to get the service period for each price."
```